public/components/Callback.js

```diff
@@ -17,14 +17,18 @@ function Callback() {
 
   const checkSuccess = useCallback(() => {
     if (
-      (permissions || process.env.PERMISSIONS_DISABLED === "true") &&
+      (permissions.length > 0 || process.env.PERMISSIONS_DISABLED === "true") &&
       Object.hasOwn(localStorage, "token")
     ) {
       setInfoMessage(
         "Everything is loaded, you should be sent to the homepage in a second.",
       );
       window.location.replace("/");
       return true;
+    } else if (permissions.length == 0) {
```

Comment:
I think double equals will yield lint error here. You can either use triple equals, or just `permissions.length` since 0 is falsey.

Reply:
Good point, I changed it